packages/transaction-controller/src/helpers/IncomingTransactionHelper.ts

```diff
@@ -191,7 +191,8 @@ export class IncomingTransactionHelper {
           (currentTx) =>
             currentTx.hash?.toLowerCase() === tx.hash?.toLowerCase() &&
             currentTx.txParams.from?.toLowerCase() ===
-              tx.txParams.from?.toLowerCase(),
+              tx.txParams.from?.toLowerCase() && 
+              currentTx.type?.toLocaleLowerCase() === tx.type?.toLocaleLowerCase(),
```

Comment:
There shouldn't be any need for `toLocaleLowerCase` here since `type` uses our `TransactionType` enum so the case won't change.